pandas/_testing.py

```diff
@@ -1181,12 +1195,17 @@ def assert_series_equal(
             check_less_precise=check_less_precise,
             check_dtype=check_dtype,
             obj=str(obj),
+            index_values=np.asarray(left.index),
         )
     elif is_extension_array_dtype(left.dtype) and is_extension_array_dtype(right.dtype):
-        assert_extension_array_equal(left._values, right._values)
+        assert_extension_array_equal(
+            left._values, right._values, index_values=np.asarray(left.index)
```

Comment:
Do you have to use asarray here or could you just pass back the DatetimeArray? Would make for nicer output if the date time values were shown

Reply:
That is a good point which I hadn't thought of.  However, it isn't quite as simple as that because we only know that the data is an extension array at this point - the index may not be.  Also, my understanding of extension array is very poor, but I thought it could be used for things other than datetimes?

The main reason though, is that the data is currently dislayed as numeric timestamps, so it seems reasonable to fix the index display at the same time as the data.  And that sounds like a seperate PR to me.  I am looking at doing it (when I get time), but as mentioned above, I need to figure out extension arrays first.

Reply:
And thanks for looking at the PR.  Still not sure if I'm supposed to click 'Resolve conversation' now.